ui/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/ValidationResults.scala

```diff
@@ -1,25 +1,28 @@
 package pl.touk.nussknacker.restmodel.validation
 
-import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
 import cats.implicits._
-import io.circe.{Decoder, Encoder, Json}
 import io.circe.generic.JsonCodec
+import io.circe.{Decoder, Encoder, Json}
 import pl.touk.nussknacker.engine.api.expression.ExpressionTypingInfo
 import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.restmodel.definition.UIParameter
 
 object ValidationResults {
 
   private implicit val typingResultDecoder: Decoder[TypingResult] = Decoder.decodeJson.map(_ => typing.Unknown)
 
+  //TODO: consider extracting additional DTO class
   @JsonCodec case class ValidationResult(errors: ValidationErrors, warnings: ValidationWarnings,
                                          variableTypes: Map[String, Map[String, TypingResult]],
+                                         parameters: Map[String, List[UIParameter]],
```

Comment:
Shouldn't we merge both `variableTypes` and `parameters` to some kind of `NodeValidationResult`?  ... and rename `variableTypes` to `inputVariableTypes` to clarify that additional variables in parameters will be added next to `inputVariableTypes`. In `parameters` there are only dynamic one and should be merged with static (from definition)? If so, maybe we should rename them to `dynamicParameterDefinitions`?

Reply:
... this wrapping type will be also useful when we will introduce inferred result type. We could use dynamic parameters + inferred result types to achieve this: https://github.com/TouK/nussknacker/issues/249#issuecomment-572517053

Reply:
Done, hope I didn't break anything on FE...

Reply:
Thanks!